util/reggen/fpv_csr.sv.tpl

```diff
@@ -56,33 +68,47 @@ module ${mod_base}_csr_assert_fpv import tlul_pkg::*;
   assign a_mask_bit[23:16] = h2d.a_mask[2] ? '1 : '0;
   assign a_mask_bit[31:24] = h2d.a_mask[3] ? '1 : '0;
 
+  bit [${addr_msb}-2:0] hro_idx; // index for exp_vals
+  bit [${addr_msb}:0]   normalized_addr;
+
+  // Map register address with hro_idx in exp_vals array.
+  always_comb begin: decode_hro_addr_to_idx
+    unique case (pend_trans[d2h.d_source].addr)
+% for hro_reg in hro_regs_list:
+      ${hro_reg.offset}: hro_idx <= ${hro_idx};
 <%
-  addr_width = rb.get_addr_width()
-  addr_msb  = addr_width - 1
-  block_size = ((rb.flat_regs[-1].offset) >> 2) + 1
+  hro_map[hro_reg.offset] = hro_idx;
+  hro_idx = hro_idx+1;
```

Comment:
Rather than update `hro_idx` here, it's probably cleaner to change the for loop to `for hro_idx, hro_reg in enumerate(hro_regs_list)`